packages/wrangler/src/type-generation/runtime/index.ts

```diff
@@ -0,0 +1,100 @@
+import { readFileSync } from "fs";
+import { writeFile } from "fs/promises";
+import { Miniflare } from "miniflare";
+import { ensureDirectoryExists } from "../../utils/log-file";
+import type { Config } from "../../config/config";
+
+const DEFAULT_OUTFILE_RELATIVE_PATH = "./.wrangler/types/runtime.d.ts";
+
+/**
+ * Generates runtime types for a Workers project based on the provided project configuration.
+ *
+ * This function is designed to be isolated and portable, making it easy to integrate into various
+ * build processes or development workflows. It handles the whole process of generating runtime
+ * types, from ensuring the output directory exists to spawning the workerd process (via Miniflare)
+ * and writing the generated types to a file.
+ *
+ * @throws {Error} If the config file does not have a compatibility date.
+ *
+ * @example
+ * import { generateRuntimeTypes } from './path/to/this/file';
+ * import { readConfig } from './path/to/config';
+ *
+ * const configPath = './wrangler.toml';
+ * const config = readConfig(configPath);
+ * const outFile = '/Users/me/my-project/dist/runtime.d.ts'
+ *
+ * // This will generate runtime types and write them to ./.wrangler/types/runtime.d.ts
+ * await generateRuntimeTypes({ config });
+ *
+ * * // This will generate runtime types and write them to /Users/me/my-project/dist/runtime.d.ts
+ * await generateRuntimeTypes({ config, outFile });
+ *
+ * @remarks
+ * - The generated types are written to a file specified by DEFAULT_OUTFILE_RELATIVE_PATH.
+ * - This could be improved by hashing the compat date and flags to avoid unnecessary regeneration.
+ */
+export async function generateRuntimeTypes({
+	config,
+	generateFn = generate,
+	outFile = DEFAULT_OUTFILE_RELATIVE_PATH,
+}: {
+	config: Config;
+	generateFn?: typeof generate;
+	outFile?: string;
+}) {
+	const { compatibility_date, compatibility_flags = [] } = config;
+
+	if (!compatibility_date) {
+		throw new Error("Config must have a compatability date.");
+	}
+
+	await ensureDirectoryExists(outFile);
+
+	const types = await generateFn({
+		compatibilityDate: compatibility_date,
+		compatibilityFlags: compatibility_flags,
+	});
+
+	await writeFile(outFile, types, "utf8");
+
+	return {
+		outFile,
+	};
```

Comment:
Why is this an object rather than just a string?

Reply:
The reason I did this was:
- if we need to extend it by returning the content too, then this way it will be backwards-compatible - i.e. require changes in fewer places, and
- it's more explicit (and therefore readable) what is actually being returned to the calling function.

I can remove it if you think it's overkill though.

Reply:
Makes sense!